e2e-tests/src/tests/scheduler.test.ts

```diff
@@ -184,18 +190,87 @@ test.describe('Scheduling', () => {
     let it = 0;
     let reason_local: string;
     while (it++ < max_it && (status == 'pending' || status == 'incomplete')) {
-      const { reason, status, analysisId } = await req.schedule(request, specification_id);
+      const { reason, status, analysisId, datasetId } = await req.schedule(request, specification_id);
       status_local = status;
       reason_local = reason;
       expect(status).not.toBeNull();
       expect(status).toBeDefined();
+      if(datasetId != null){
+        dataset_id = datasetId
+      }
```

Comment:
Why is there a null check before assigning `dataset_id`?

Reply:
because datasetId is null until scheduling is finished.

Reply:
Right, but what's the harm in assigning dataset_id to a null value?

Reply:
I don't remember at this point. I have removed the if.